src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -50,6 +52,21 @@ public static class WebScriptHostBuilderExtension
 
                     loggingBuilder.AddWebJobsSystem<SystemLoggerProvider>();
                     loggingBuilder.Services.AddSingleton<ILoggerProvider, UserLogMetricsLoggerProvider>();
+                    loggingBuilder.Services.AddSingleton<ILoggerProvider>(services =>
+                    {
+                        IEnvironment environment = services.GetService<IEnvironment>();
+                        IScriptWebHostEnvironment hostEnvironment = services.GetService<IScriptWebHostEnvironment>();
+
+                        if (!hostEnvironment.InStandbyMode &&
+                            !string.IsNullOrEmpty(environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteHostName)))
```

Comment:
Is this check for host name to ensure that we're running in AppService? There are EnvironmentExtensions for that - IsAppServiceEnvironment/IsLinuxAppServiceEnvironment

Reply:
Without WEBSITE_HOSTNAME, we can't use this logger as we'd have no resourceId, which is required. I'm checking explicitly for that rather than inferring it'll be there with a different API.